pkg/diagnose/connectivity/core_endpoint.go

```diff
@@ -100,6 +100,20 @@ func Diagnose(diagCfg diagnosis.Config) []diagnosis.Diagnosis {
 				}
 				diagnoses = append(diagnoses, d)
 			}
+
+			// Send a request to the flare endpoint
```

Comment:
Is it possible to factor this whole code block you added within `endpointsInfo` to avoid duplication ?
Ie. update `sendHTTPRequestToEndpoint` to check if the method is a `HEAD` and if so use the single redirect client

Reply:
Yes I will work on it and commit changes